tools/ldb_cmd.cc

```diff
@@ -3351,7 +3351,18 @@ void DBFileDumperCommand::DoCommand() {
   }
   // remove the trailing '\n'
   manifest_filename.resize(manifest_filename.size() - 1);
-  std::string manifest_filepath = db_->GetName() + "/" + manifest_filename;
+  std::string manifest_filepath = db_->GetName();
+  // Correct concatenation of filepath and filename:
+  // Add slash '/' in between if necessary, or check that
+  // there is no double slashes when concatenation happens.
+  if (manifest_filepath.size() > 0 && manifest_filepath.back() != '/' &&
+      manifest_filename.front() != '/') {
+    manifest_filepath += "/";
+  } else if (manifest_filepath.size() > 0 && manifest_filepath.back() == '/' &&
+             manifest_filename.front() == '/') {
+    manifest_filepath.pop_back();
+  }
+  manifest_filepath += manifest_filename;
```

Comment:
Is this different than NormalizePath?  file/filename.{cc,h}

Reply:
Thanks this is very useful I wasn't aware of this function - I made an edit such that all the situations are handled, including if neither manifest_filepath nor manifest_filename have '/' (because then the NormalizePath function wouldn't be able to add a missing '/').

Reply:
As it is coded (without NormalizePath), I am not sure this will necessarily work on Windows.

Reply:
Apologies, it took me a couple minutes to test my edit locally before committing - I just pushed the update a minute ago if you want to take a look at the edit (with NormalizePath).